xarray/tests/test_accessor_dt.py

```diff
@@ -148,6 +148,118 @@ def test_rounders(self):
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        nt = 100
+        data = np.random.rand(10, 10, nt)
+        lons = np.linspace(0, 11, 10)
+        lats = np.linspace(0, 20, 10)
+        self.times = pd.date_range(start="2000/01/01", freq="H", periods=nt)
+        self.times = self.times - self.times[0]
+
+        self.data = xr.DataArray(
+            data,
+            coords=[lons, lats, self.times],
+            dims=["lon", "lat", "time"],
+            name="data",
+        )
+
+        self.times_arr = np.random.choice(self.times, size=(10, 10, nt))
+        self.times_data = xr.DataArray(
+            self.times_arr,
+            coords=[lons, lats, self.times],
+            dims=["lon", "lat", "time"],
+            name="data",
+        )
+
+    def test_field_access(self):
+        days = xr.DataArray(
+            self.times.days, name="days", coords=[self.times], dims=["time"]
+        )
+        seconds = xr.DataArray(
+            self.times.seconds, name="seconds", coords=[self.times], dims=["time"]
+        )
+        microseconds = xr.DataArray(
+            self.times.microseconds,
+            name="microseconds",
+            coords=[self.times],
+            dims=["time"],
+        )
+        nanoseconds = xr.DataArray(
+            self.times.nanoseconds,
+            name="nanoseconds",
+            coords=[self.times],
+            dims=["time"],
+        )
+
+        assert_equal(days, self.data.time.dt.days)
+        assert_equal(seconds, self.data.time.dt.seconds)
+        assert_equal(microseconds, self.data.time.dt.microseconds)
+        assert_equal(nanoseconds, self.data.time.dt.nanoseconds)
+
+    def test_not_datetime_type(self):
+        nontime_data = self.data.copy()
+        int_data = np.arange(len(self.data.time)).astype("int8")
+        nontime_data["time"].values = int_data
+        with raises_regex(TypeError, "dt"):
+            nontime_data.time.dt
+
+    @requires_dask
+    def test_dask_field_access(self):
+        import dask.array as da
+
+        days = self.times_data.dt.days
+        seconds = self.times_data.dt.seconds
+        microseconds = self.times_data.dt.microseconds
+        nanoseconds = self.times_data.dt.nanoseconds
+        floor = self.times_data.dt.floor("D")
+        ceil = self.times_data.dt.ceil("D")
+        round = self.times_data.dt.round("D")
+
+        dask_times_arr = da.from_array(self.times_arr, chunks=(5, 5, 50))
+        dask_times_2d = xr.DataArray(
+            dask_times_arr, coords=self.data.coords, dims=self.data.dims, name="data"
+        )
+        dask_days = dask_times_2d.dt.days
+        dask_seconds = dask_times_2d.dt.seconds
+        dask_microseconds = dask_times_2d.dt.microseconds
+        dask_nanoseconds = dask_times_2d.dt.nanoseconds
+        dask_floor = dask_times_2d.dt.floor("D")
+        dask_ceil = dask_times_2d.dt.ceil("D")
+        dask_round = dask_times_2d.dt.round("D")
+
+        # Test that the data isn't eagerly evaluated
```

Comment:
import `raise_if_dask_computes` from `test_dask` and wrap the previous paragraph so that we're sure compute isn't called when `.dt` is accessed.